### PR TITLE
fix(store-helper): do not list canceled tasks

### LINKED ISSUE #1060: hide 'cancelled' statuses
### Steps to reproduce
1. create a task
2. set status of task to 'cancelled'

### Expected behaviour
The task should be removed from view. Ideally to a separate section, say, **Cancelled Tasks** like Completed Tasks

### Actual behaviour
Nothing. The task just stays there as if nothing happened.

### Server configuration

**Operating system**: Debian 10 "Buster"

**Web server:** Apache 2.4.38

**Database:** MariaDB 10.3.22

**PHP version:** 7.3.14

**Nextcloud version:** 18.0.4 ( git )

**Tasks version:** 0.13.1

**Updated from an older Nextcloud or fresh install:** Not Applicable but I've been upgrading via git since v11

**Signing status:** `Integrity checker has been disabled. Integrity cannot be verified.`

**List of activated apps:**

<details>
<summary>Enabled Apps</summary>

```
  - accessibility: 1.4.0
  - apporder: 0.9.0
  - audioplayer: 2.10.0
  - bookmarks: 3.0.13
  - calendar: 2.0.3
  - cloud_federation_api: 1.1.0
  - comments: 1.8.0
  - contacts: 3.3.0
  - dav: 1.14.0
  - deck: 1.0.1
  - documentserver_community: 0.1.5
  - federatedfilesharing: 1.8.0
  - federation: 1.8.0
  - files: 1.13.1
  - files_frommail: 0.4.0
  - files_markdown: 2.2.0
  - files_mindmap: 0.0.21
  - files_rightclick: 0.15.2
  - files_sharing: 1.10.1
  - files_trashbin: 1.8.0
  - files_versions: 1.11.0
  - lookup_server_connector: 1.6.0
  - mail: 1.3.5
  - maps: 0.1.6
  - metadata: 0.11.1
  - music: 0.14.1
  - news: 14.1.7
  - notes: 3.3.1
  - notifications: 2.6.0
  - oauth2: 1.6.0
  - ocsms: 2.1.7
  - onlyoffice: 4.1.4
  - ownpad: 0.6.14
  - passman: 2.3.5
  - passwords: 2020.5.0
  - phonetrack: 0.6.3
  - photos: 1.0.0
  - polls: 1.4.3
  - privacy: 1.2.0
  - provisioning_api: 1.8.0
  - recommendations: 0.6.0
  - settings: 1.0.0
  - sharebymail: 1.8.0
  - socialsharing_diaspora: 2.0.0
  - socialsharing_email: 2.0.0
  - socialsharing_twitter: 2.0.0
  - spreed: 8.0.9
  - support: 1.1.0
  - systemtags: 1.8.0
  - talk_simple_poll: 1.1.1
  - tasks: 0.13.1
  - text: 2.0.0
  - theming: 1.9.0
  - timetracker: 0.0.42
  - twofactor_backupcodes: 1.7.0
  - twofactor_gateway: 0.16.0
  - twofactor_nextcloud_notification: 2.3.0
  - twofactor_totp: 4.1.3
  - twofactor_u2f: 5.1.0
  - updatenotification: 1.8.0
  - viewer: 1.2.0
  - workflowengine: 2.0.0
```
</details>

**Nextcloud configuration:**

<details>
<summary>System Configuration</summary>

```
{
    "system": {
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost",
            "cloud.shinenelson.com"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "18.0.4.2",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "theme": "",
        "loglevel": 1,
        "maintenance": false,
        "overwrite.cli.url": "***REMOVED SENSITIVE VALUE***",
        "app_install_overwrite": [
            "user_pwauth"
        ],
        "app.mail.imaplog.enabled": true,
        "app.mail.smtplog.enabled": true,
        "app.mail.smtp.timeout": 20,
        "app.mail.verify-tls-peer": false,
        "htaccess.RewriteBase": "\/",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_sendmailmode": "smtp",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "587",
        "mail_smtpsecure": "tls",
        "mail_smtptimeout": "20",
        "mail_smtpauth": 1,
        "memcache.local": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0
        }
    }
}
```
</details>

**Are you using external storage:** no

**Are you using encryption:** yes

**Are you using an external user-backend:** no

### Client configuration
**Browser:** Firefox 76.0.1

**Operating system:** Ubuntu 20.04

**CalDAV-clients:** Not applicable

### PR DESCRIPTION
**Pull request by dargmuesli**
Resolves #1060.

### CODE REVIEW

**Comment by raimund-schluessler** [comment]
Thanks for the contribution!

Without having run the code, I think cancelled tasks are shown nowhere now. Please show them under completed, as discussed in #1060. Adjusting this line https://github.com/nextcloud/tasks/pull/1418/files#diff-1cba3917082f42bccf54c8c66fff865c2da4ce850b0a53665d84b6fc45f5529aL41 should be enough, I think.

**Comment by dargmuesli** [comment]
Sure, done! :)

File: src/store/storeHelper.js [hunk 1]
```diff
@@ -2,6 +2,7 @@
  * Nextcloud - Tasks
  *
  * @author Raimund Schlüßler
+ * @copyright 2020 Jonas Thelemann <e-mail@jonas-thelemann.de>
```
Comment (shinenelson) [inline src/store/storeHelper.js:5]: tiny nitpick : you probably meant to add 2021 as the copyright year.

Comment (dargmuesli) [reply src/store/storeHelper.js]: Seems like the missing fireworks made me miss new year. Thanks for the hint! Fixed.

Comment (shinenelson) [reply src/store/storeHelper.js]: the only firework I can give you is :fireworks: :sparkler:  I hope this would suffice :slightly_smiling_face:

**Comment by dargmuesli** [comment]
@raimund-schluessler when do you think this can be reviewed? Just wondering.

**Comment by raimund-schluessler** [comment]
> 
> 
> @raimund-schluessler when do you think this can be reviewed? Just wondering.

I will try to have a look at the weekend.

I think what we should also do, is show canceled tasks similar as completed tasks are shown (with strike-through, but maybe with a cross, instead of checkmark). But this can be done in a follow-up PR.

**Comment by raimund-schluessler** [comment]
I had a look and it works as expected. But there are two things to improve:

- It has to be obvious that the task is canceled. As said by strike through and a cross instead of a checkmark.
- I think it makes sense to set the subtasks of a canceled task to canceled as well, just as we do it with completed tasks. Means when we remove the canceled status of a task, parent tasks need to be uncanceled as well (also as we do it for completed value).

@dargmuesli Do you to take care of this, or should we do it in a follow up PR? Since you are not member of this repository, it is a bit more difficult to collaborate on the same PR. Do you mind if we add you to the repo?

**Comment by dargmuesli** [comment]
Sorry, I currently don't have the time, but I agree, that would be cool. Let's use a separate PR for that.

**Comment by raimund-schluessler** [comment]
@dargmuesli I hope you don't mind that I'll move this PR to a follow-up PR #1553 so I can push more changes to the branch before merging.

**Pull request closed without merging**